tools/hashes_collection.sh

```
@@ -1,18 +1,60 @@
 #!/bin/bash
-# set of shell commands to hash the set of data objects and hierarchy in a collection
-# To compare the content of 2 collections, update the shell variable COLLECTION and rerun the query line
-# explanation:
-# 1. we select logical path and data object size for each data object within the collection tree
-# 2. we remove the prefix (collection name) so that the logical name starts within the collection (needed for comparison)
-# 3. we sort the pathnames, using a locale that ensures all characters have a different sorting order
-# bash commands:
-
-# Check if COLLECTION argument is provided
-if [ -z "$1" ]; then
-    echo "Error: Collection path must be provided as an argument." >&2
-    echo "Usage: $0 <collection-path>" >&2
-    exit 1
+
+#set of shell commands to hash the set of data objects and hierarchy in a collection
+#To compare the content of 2 collections, update the shell variable COLLECTION and rerun the query line
+#explanation:
+#1. we select logical path and data object size for each data object within the collection tree
+#2. we remove the prefix (collection name) so that the logical name starts within the collection (needed for comparison)
+#3. we sort the pathnames, using a locale that ensures all characters have a different sorting order
+
+#bash commands:
+
+DETAILED=0
+DEBUG=0
+
+# Parse arguments
+for arg in "$@"; do
+    case "$arg" in
+        --detailed) DETAILED=1 ;;
+        --debug) DEBUG=1 ;;
+        *) if [[ -z "$COLLECTION" ]]; then COLLECTION="${arg%/}"; fi ;;
+    esac
+done
+
+if [ -z "$COLLECTION" ]; then
+   echo "Usage: bash $0 <collectionName> [--detailed] [--debug]"
```

The current version of the script silently ignores the collection name argument if a `COLLECTION` variable is present in the environment. Does this behaviour facilitate a particular use case? If so, I suggest making it clear that the argument is being ignored by e.g. printing a message that the collection name from the environment is used instead and the command-line argument is ignored. If we don't have a specific reason for this behaviour, we can simplify by just always using the command-line argument.

---

Changed as suggested.